File: 20210321-java-shoppingbasket/src/test/java/com/katas/ShoppingBasketTest.java
package com.katas;

import org.junit.Test;

import java.util.Arrays;

import static org.assertj.core.api.Assertions.assertThat;

public class ShoppingBasketTest {

    private ShoppingBasket shoppingBasket;

    @Test
    public void should_have_a_total_of_zero_when_empty() {
        shoppingBasket = new ShoppingBasket();
        assertThat(shoppingBasket.getTotal()).isZero();
    }

    @Test
    public void should_calculate_total_for_single_item() {
        shoppingBasket = new ShoppingBasket(Arrays.asList(new Item(100, 1)));
        assertThat(shoppingBasket.getTotal()).isEqualTo(100);
    }
}

File: 20210321-kotlin-shoppingbasket/src/test/kotlin/com/katas/ShoppingBasketTest.kt
package com.katas

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class ShoppingBasketTest {
    @Test
    fun `should have total zero when is empty`() {
        val shoppingBasket = ShoppingBasket()
        assertThat(shoppingBasket.total).isZero()
    }
}
